roachtest: tpcc-1000 roachtests have failing checks

Some roachtests that involve running tpcc-1000 are failing the tpcc checks at the end, e.g.:

```
		I190114 15:41:47.835436 1 workload/tpcc/tpcc.go:290  check 3.3.2.1 took 162.495391ms
		Error: check failed: 3.3.2.1: 12 rows returned, expected zero
```
The tests affected by this:

- #33624 (schemachange/index/tpcc-1000)
- #33335 (schemachange/indexrollback/tpcc-1000)
- #33151 (scrub/index-only/tpcc-1000)
- #33149 (scrub/all-checks/tpcc-1000)
- https://github.com/cockroachdb/cockroach/issues/32813#issuecomment-453920621 (cdc/tpcc-1000) is potentially related

https://teamcity.cockroachdb.com/viewLog.html?buildId=1085451&buildTypeId=Cockroach_Nightlies_WorkloadNightly&tab=buildResultsDiv seems to be the first test failure of this kind. I reproduced this failure on `scrub/index-only/tpcc-1000` on a separate roachprod cluster, and ran SCRUB on the entire tpcc database, which did not turn up any anomalies.

It's not clear whether this is a schema change problem, a problem with the tests, or something else. Some next steps would be to determine whether this occurs when just running tpcc-1000 by itself, and look at what the anomalous rows are in the results of the check query.

The first failure on `master` is in:

```
--- FAIL: scrub/index-only/tpcc-1000 (10829.34s)
	test.go:743,cluster.go:1226,tpcc.go:130,scrub.go:58: /Users/tschottdorf/go/bin/roachprod run tobias-1549443202-scrub-index-only-tpcc-1000:5 -- ./workload check tpcc --warehouses=1000 {pgurl:1} returned:
		stderr:
		I190206 11:53:20.623699 1 workload/tpcc/tpcc.go:288  check 3.3.2.1 took 2.090668716s
		Error: check failed: 3.3.2.1: 9 rows returned, expected zero
		Error:  exit status 1

		stdout:
		: exit status 1
```

the second `master` instance is still running. The two fixed instances have already made it past check 3.3.2.1 🙌 and probably even all the checks (the test is "waiting" which I assume is the scrub part).

So assuming the second master run also fails, I'm fairly comfortable that that was the bug.

and here's the second failure on `master`:

```
--- FAIL: scrub/index-only/tpcc-1000 (11138.28s)
	test.go:743,cluster.go:1226,tpcc.go:130,scrub.go:58: /Users/tschottdorf/go/bin/roachprod run tobias-1549443223-scrub-index-only-tpcc-1000:5 -- ./workload check tpcc --warehouses=1000 {pgurl:1} returned:
		stderr:
		I190206 11:58:48.216353 1 workload/tpcc/tpcc.go:288  check 3.3.2.1 took 3.921505659s
		Error: check failed: 3.3.2.1: 7 rows returned, expected zero
		Error:  exit status 1

		stdout:
		: exit status 1
```

--- a/pkg/storage/replica_raft.go
+++ b/pkg/storage/replica_raft.go
@@ -1927,7 +1927,7 @@ func (r *Replica) processRaftCommand(
 				log.Fatalf(ctx, "proposal must return either a reply or an error: %+v", proposal)
 			}
 			response.Intents = proposal.Local.DetachIntents()
-			response.EndTxns = proposal.Local.DetachEndTxns(response.Err != nil)
+			response.EndTxns = proposal.Local.DetachEndTxns(pErr != nil)
 			if pErr == nil {
 				lResult = proposal.Local
 			}

